backend/routes/maps.js

```diff
@@ -21,15 +21,41 @@ const router = express.Router({mergeParams: true});
 const middlewares = require("../middlewares/middlewares");
 const systemLogger = require("../logger.js").systemLogger;
 const httpsGet = require('../libs/httpsReq');
+const config = require("../config");
+const User = require('../models/user');
 
+router.get("/:model/maps/", listMaps);
 router.get("/:model/maps/osm/:zoomLevel/:gridx/:gridy.png", getOSMTile);
+router.get("/:model/maps/here/:zoomLevel/:gridx/:gridy.png", middlewares.isHereEnabled, getHereMapsTile);
+router.get("/:model/maps/hereaerial/:zoomLevel/:gridx/:gridy.png", middlewares.isHereEnabled, getHereAerialMapsTile);
+router.get("/:model/maps/heretraffic/:zoomLevel/:gridx/:gridy.png", middlewares.isHereEnabled, getHereTrafficTile);
+router.get("/:model/maps/heretrafficflow/:zoomLevel/:gridx/:gridy.png", middlewares.isHereEnabled, getHereTrafficFlowTile);
+router.get("/:model/maps/herebuildings/:lat/:long/tile.json", middlewares.isHereEnabled, getHereBuildingsFromLongLat);
 
-function getOSMTile(req, res, next){
-	//TODO: we may want to ensure the model has access to tiles
-//	const url = "https://a.tile.openstreetmap.org/" + req.params.zoomLevel + "/" + req.params.gridx + "/" + req.params.gridy + ".png";
-	const domain = "a.tile.openstreetmap.org";
-	const uri = "/" + req.params.zoomLevel + "/" + req.params.gridx + "/" + req.params.gridy + ".png"
-	systemLogger.logInfo("Fetching osm map tile: " + uri);
+function listMaps(req, res, next) {
+	const user = (req.session.user) ? req.session.user.username : "";
+
+	let maps = [
+		{ name: "Open Street Map", layers: [ { name: "Map Tiles", source: "OSM" } ] }
+	];
+
+	User.isHereEnabled(user).then((hereEnabled) => {
```

Comment:
Again, this needs to be the account the model belongs to, not the user who logged in

Reply:
Also for this specific screenshot, the tiles can be much larger. Ideally, the algorithm should somehow try to fill in the most space with a set number of tiles if that makes sense.